### ISSUE TITLE
[Anon] Bug report: Fallout_Wasteland_Warfare.gstz

### ISSUE THREAD

**Issue by BSDataAnon** [BSData/fallout-wasteland-warfare#12]
**File:** Fallout_Wasteland_Warfare.gstz

**BattleScribe version:** 2.03.20

**Platform:** Android

**Dropbox:** Yes

**Description:** creating & loading a fallout roster takes 2+ minutes on multiple devices (newer high end phone and on a Chromebook).
Looking into the gst file there is an enormous amount of conditions related to the hidden status of items, weapons etc.
This is a massive performance drain. Just as a test I removed ALL condition groups and retested with the same scenario. Load time improved to 4 seconds rather than 2+ minutes!
Needless to say I prefer having 4 second load time rather than a precise list of items the character is allowed to take. Please consider improving this aspect of the datafiles for the next update. Other than that I'm massively grateful for all the data entry performed by you. Really helps out once the roster has loaded.

**Comment by Cosworth32** [comment]
Sorry - forgot to login first. I'm the author of the submission above

**Comment by Aroksak** [comment]
I think I know how to optimize it, just need to find some time to implement it. Hope I will dive into it on the weekend.

**Comment by Aroksak** [comment]
Well, I've rearranged the inner structure greatly, but looks like it didn't improve a thing, it still takes years to create a roaster on my phone. I will keep trying, any suggestions are welcome. Removing all conditions, though, makes this whole thing kinda useless IMO.

**Comment by Cosworth32** [comment]
Yeah I noticed the new version was still very slow to load. I think the logic concerning who gets what weapons/armor/gear etc. is upside down now. When you load a roster all selection are prepared including the selectable items. For a faction with 15 profiles you are currently doing at least 500 IF-Then statements per profile. This quickly adds up. Survivors with 28 profiles do around 14.000 IF statement checks when loading the roster.
As the goal must be to limit selections according to play mode, I think you need to do the if statements on the profiles and not on the items.
For AI mode this is simple as most only have a few items per default. Battle mode is tediously detailed per profile and settlement mode is basically the simplest as almost everything is allowed.
I know it is a lot of work redoing this, but at least it can be done by faction so we could work in parallel if you'd like some help. I could do a faction fully and let you know how that performs in comparison?

**Comment by Aroksak** [comment]
Well, I tried to revert the logic --- I removed conditions from equipment and transferred them to units. But as you could see it didn't help much, or I did something wrong. In the version I released today (will take some time to appear in apps) I separated different factions to different files. Performance improvement on PC is noticeable, but not too great. By tomorrow we should be able to test it on mobiles.
Of course help is much appreciated, thank you!

**Comment by Cosworth32** [comment]
You did it!. It works much faster now. 7-8 times faster loading time in battle mode and a little faster in narrative mode.
Must be the unhiding that really drains performance

**Comment by Cosworth32** [comment]
Another big performance drain are the mods. There's 50+ of them added for every item that can use mods. Just separating them in mod groups (e.g. pistol mods, melee mods etc) would yield substantial additional benefit.
